iphone/Classes/CMHelper.m

```diff
@@ -135,18 +135,18 @@ +(NSDictionary*)dictionaryFromMotionActivity:(CMMotionActivity*)motionActivity
 +(NSDictionary*)dictionaryFromPedometerData:(CMPedometerData*)pedometerData
 {
     NSMutableDictionary *dict = [NSMutableDictionary dictionaryWithDictionary:@{
-        @"startDate": (!pedometerData) ? [NSNull null] : [TiUtils UTCDateForDate:pedometerData.startDate],
-        @"endDate": (!pedometerData) ? [NSNull null] : [TiUtils UTCDateForDate:pedometerData.endDate],
-        @"numberOfSteps": (!pedometerData) ? [NSNull null] : NUMINT(pedometerData.numberOfSteps),
-        @"distance": (!pedometerData) ? [NSNull null] : NUMINT(pedometerData.distance),
-        @"floorsAscended": (!pedometerData) ? [NSNull null] : NUMINT(pedometerData.floorsAscended),
-        @"floorsDescended": (!pedometerData) ? [NSNull null] : NUMINT(pedometerData.floorsDescended)        
+        @"startDate": (!pedometerData.startDate) ? [NSNull null] : [TiUtils UTCDateForDate:pedometerData.startDate],
+        @"endDate": (!pedometerData.endDate) ? [NSNull null] : [TiUtils UTCDateForDate:pedometerData.endDate],
+        @"numberOfSteps": (!pedometerData.numberOfSteps) ? [NSNull null] : pedometerData.numberOfSteps,
+        @"distance": (!pedometerData.distance) ? [NSNull null] : pedometerData.distance,
+        @"floorsAscended": (!pedometerData.floorsAscended) ? [NSNull null] : pedometerData.floorsAscended,
+        @"floorsDescended": (!pedometerData.floorsDescended) ? [NSNull null] : pedometerData.floorsDescended
```

Comment:
Not sure how this solution would change the incorrect number of steps returned. The problem was not that something was returned null or non-null, but the specific value being different. I highly doubt that this will change something, but please describe your solution.